src/NumericalAlgorithms/Interpolation/BarycentricRational.hpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <unistd.h>  // IWYU pragma: keep
+#include <vector>
+
+namespace PUP {
+class er;
+}  // namespace PUP
+
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief Contains classes and functions for interpolation
+ */
+namespace intrp {
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief A barycentric rational interpolation class
+ *
+ * The class builds a barycentric rational interpolant of a specified order
+ * using the `x_values` and `y_values` passed into the constructor.
+ * Barycentric interpolation requires \f$3N\f$ storage, and costs
+ * \f$\mathcal{O}(Nd)\f$ to construct, where \f$N\f$ is the size of the x- and
+ * y-value vectors and \f$d\f$ is the order of the interpolant. The evaluation
+ * cost is \f$\mathcal{O}(N)\f$ compared to \f$\mathcal{O}(d)\f$ of a spline
+ * method, but constructing the barycentric interpolant does not require any
+ * derivatives of the function to be known.
+ *
+ * The interpolation function is
+ *
+ * \f[
+ *   \mathcal{I}(x)=\frac{\sum_{i=0}^{N-1}w_i y_i /
+ *   (x-x_i)}{\sum_{i=0}^{N-1}w_i/(x-x_i)}
+ * \f]
+ *
+ * where \f$w_i\f$ are the weights. The weights are computed using
+ *
+ * \f[
+ *   w_k=\sum_{i=k-d\\0\le i\le N-d}^{k}(-1)^{k}
```

Comment:
\le N-d should be < N-d (see response to comments in .cpp file)

Reply:
Hi Saul, that's not the part we are confused on. The part we are confused on is that (at least in my copy of NR) the weights are given by `w_k = sum_{i ...}^{...}(-1)^k ...` but the actual code, at least naively, reads as `(-1)^i`. Do you know what we're missing?

Reply:
Yes! There is a typo in NR eqn. 3.4.10. `(-1)^k` should be `(-1)^i`
The code is correct.